strictdoc/backend/sdoc/processor.py

```diff
@@ -64,10 +67,30 @@ def process_section(self, section: Section):
                 )
 
         if section.parent.ng_level is None:
-            SDocParsingProcessor._resolve_parents(section)
+            self._resolve_parents(section)
         section.ng_level = section.parent.ng_level + 1
         assert section.ng_level > 0
 
+    def process_include(self, include: IncludeFromFile):
+        # pylint: disable=import-outside-toplevel
+        from strictdoc.backend.sdoc.reader import (
+            SDIReader,
+        )  # can't import globally or else module loop ensues
+
+        if include.parent.ng_level is None:
+            self._resolve_parents(include)
+        self.parse_context.current_include_parent = include.parent
+
+        reader = SDIReader()
+        fragment = reader.read_from_file(
+            file_path=include.file, context=self.parse_context
+        )
+        assert isinstance(fragment, Fragment)
+
+        include.parent.free_texts.extend(fragment.free_texts)
```

Comment:
I am not sure if this code is fully stable. Considering the case when a fragment is included somewhere between the sections like:

```
free text
section
section
...
fragment
...
section
section
```

One option to eliminate this case will be to exclude free text from `FRAGMENT` and only allow them within the sections.

Reply:
I will solve it by excluding the free texts.